rts/lra/coordinator/src/main/java/io/narayana/lra/coordinator/api/JaxRsApplication.java

```diff
@@ -0,0 +1,40 @@
+package io.narayana.lra.coordinator.api;
+
+import io.narayana.lra.LRAConstants;
+import org.eclipse.microprofile.openapi.annotations.Components;
+import org.eclipse.microprofile.openapi.annotations.OpenAPIDefinition;
+import org.eclipse.microprofile.openapi.annotations.enums.ParameterIn;
+import org.eclipse.microprofile.openapi.annotations.enums.SchemaType;
+import org.eclipse.microprofile.openapi.annotations.headers.Header;
+import org.eclipse.microprofile.openapi.annotations.info.Contact;
+import org.eclipse.microprofile.openapi.annotations.info.Info;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+import org.eclipse.microprofile.openapi.annotations.parameters.Parameter;
+import org.eclipse.microprofile.openapi.annotations.tags.Tag;
+
+import javax.ws.rs.ApplicationPath;
+import javax.ws.rs.core.Application;
+
+@OpenAPIDefinition(
+    info = @Info(title = "LRA Coordinator", version = LRAConstants.CURRENT_API_VERSION_STRING,
+        contact = @Contact(name = "Narayana", url = "https://narayana.io")),
+    tags = @Tag(name = "LRA Coordinator"),
+    components = @Components(
+        schemas = {
+            @Schema(name = "LRAApiVersionSchema",
+                description = "Format is `major.minor`, both components are required, they are to be numbers",
+                type = SchemaType.STRING, pattern = "^\\d+\\.\\d+$", example = "1.0")
+        },
+        parameters = {
+            @Parameter(name = LRAConstants.NARAYANA_LRA_API_VERSION_HEADER_NAME, in = ParameterIn.HEADER,
+                description = "Narayana LRA API version", schema = @Schema(ref = "LRAApiVersionSchema"))
+        },
+        headers = {
+            @Header(name = LRAConstants.NARAYANA_LRA_API_VERSION_HEADER_NAME, description = "Narayana LRA API version",
+                schema = @Schema(ref = "LRAApiVersionSchema"))
+        }
+    )
+)
+@ApplicationPath("/")
+public class JaxRsApplication extends Application {
```

Comment:
Am I right to see that a big part of the change is to move the `Coordinator` from extending `Application` to this new class? Does that somehow prevent the duplicate `/lra-coordinator` path? And if so, please can you share why that is?

Reply:
Yes, correct. But TBH I am not totally sure. It is inside of the smallrye openapi processing into which I didn't dig too much.

Reply:
Is there somewhere good we can ask for some information to help clarify if this is what works? I notice you have the addition of the ApplicationPath("/") in here, I wonder if it helps somehow. I wonder what would happen if many of the changes were reverted here and instead the existing `Coordinator` class had @ApplicationPath("/") instead of the previous @Path(COORDINATOR_PATH_NAME)

Reply:
Sorry, PTO. I don't think that it would help. This is the recommended Quarkus setup which I would like to preserve. Extending Application from the resource class is not a typical ways in JAX-RS. But if you want I will try to find out why it works this way, but I just don't see the value in it.

Reply:
Please will you share a reference to where it's recommended to separate the Application and the resource class?

Reply:
https://quarkus.io/guides/resteasy#whats-different-from-jakarta-ee-development and https://quarkus.io/guides/openapi-swaggerui#providing-application-level-openapi-annotations. Also I would like to note that this still passes all TCK tests so there is no functional change in this PR.

Reply:
Do you know if separating these classes is appropriate in WildFly too or is the information you are referring to specific to Quarkus?

Reply:
@xstefank mainly just by adding the ApplicationPath annotation and setting it to "/" (as you added it to the new Applications below) but in the existing Application - I find that (on Quarkus) the openapi renders fine then on "/lra-coordinator": https://github.com/jbosstm/narayana/compare/jbosstm:master...tomjenkinson:JBTM-3596?expand=1

That being said, on WildFly the coordinator is currently deploying on "/lra-coordinator/lra-coordinator" so I am thinking that with this change it could be that the application will now be on "/lra-coordinator" so this will be a public API change.